src/SAWScript/Proof.hs

```diff
@@ -902,25 +901,21 @@ data Theorem =
 data TheoremDB =
   TheoremDB
   -- TODO, maybe this should be a summary or something simpler?
-  { theoremMap :: IORef (Map.Map TheoremNonce Theorem)
+  { theoremMap :: Map.Map TheoremNonce Theorem
   }
 
-newTheoremDB :: IO TheoremDB
-newTheoremDB = TheoremDB <$> newIORef mempty
+newTheoremDB :: TheoremDB
+newTheoremDB = TheoremDB mempty
```

Comment:
I'd propose renaming this function to something like "`emptyTheoremDB`". The "`new`-" prefix would lead me to believe that this function is monadic, which is no longer the case.

Reply:
Good call, I will make that change.